bee-message/src/payload/data.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use bee_packable::{error::{PackPrefixError, UnpackPrefixError}, Packable, Packer, PackError, Unpacker, UnpackError, VecPrefix};
+
+use core::{fmt, convert::Infallible};
+
+#[derive(Debug)]
+pub enum DataPackError {
+    InvalidPrefixLength,
+}
+
+impl From<PackPrefixError<Infallible, u32>> for DataPackError {
+    fn from(_: PackPrefixError<Infallible, u32>) -> Self {
+        Self::InvalidPrefixLength
+    }
+}
+
+impl fmt::Display for DataPackError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::InvalidPrefixLength => write!(f, "Invalid prefix length for data"),
```

Comment:
Just a random thought: Maybe we should have some rules about the format of the error messages in the whole project iirc rust uses "no capitalization and no dot at the end". cc @thibault-martinez.

Reply:
No capitalization makes me cry 😢